templates/prometheus.go

```diff
@@ -35,6 +35,14 @@ var resourceSelector = metav1.LabelSelector{
 	},
 }
 
+var namespaceSelector = metav1.LabelSelector{
+	MatchExpressions: []metav1.LabelSelectorRequirement{{
+		Key:      "kubernetes.io/metadata.name",
+		Operator: metav1.LabelSelectorOpExists,
+		Values:   []string{},
+	}},
+}
+
```

Comment:
Why not?
```suggestion
var namespaceSelector = metav1.LabelSelector{}
```

Reply:
I tried doing that but the `ServiceMonitorNamespaceSelector`, `PodMonitorNamespaceSelector` and `RuleNamespaceSelector` treat is as an empty value. So only the own namespace is selected.

Reply:
- `..opExist` is also a valid identifier with no value but a valid key
- not sure why we want to have empty struct (just curious about comment reasoning)

Reply:
> ..opExists is also a valid identifier with no value but a valid key

In the go API doc, this is mentioned `If the operator is Exists or DoesNotExist, the values array must be empty`. So I used an empty array instead of no value. I will try again with passing no value.

> not sure why we want to have empty struct (just curious about comment reasoning)

I guess Ohad mentioned it because of this `An empty label selector matches all objects. A null label selector matches no objects.`. This statement is mentioned in the go API doc of the label selector.
https://pkg.go.dev/k8s.io/apimachinery@v0.26.1/pkg/apis/meta/v1#LabelSelector

Reply:
> I will try again with passing no value.
- ack, iirc, I used that during pricing
- thanks for the other ref

Reply:
@leelavg Worked without passing value, thanks!